projects/atomic-swap-service/headless-wallet.md

```diff
@@ -0,0 +1,118 @@
+[acceptance-criteria]: #acceptance-criteria
+
+- Any initialized wallet on the headless app should be able to participate on atomic swap proposals managed by the Atomic Swap Service
+- Each wallet's end users should be able to register proposals to be listened to, and receive real-time change notifications for them
+- There should be no impact on the current atomic swap workflow based on string exchanges
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The headless wallet, in a similar fashion as [what is proposed on the Desktop Wallet](https://github.com/HathorNetwork/hathor-wallet/pull/361/files#diff-81b333677b8eb7fcc977d225072c1c10453c0aee095f9458885d7a553ef3579d), will have a `listnenedProposals` map in memory for each of the initialized wallets and will keep track of their identifiers, passwords and real-time updates.
+
+### Example use case 1
+A user for the initialized wallet `swap-website-wallet` calls the `registerProposal` endpoint with an identifier `prop-1` and password `123` that it has received from another one of the proposal's participant. This request will add `prop-1` to the local `listenedProposals` object in memory.
+
+A `fetch` call can be made right after, informing the desired `prop-1` identifier, to retrieve its latest state on the service backend and, after changes are made,  `update` requests can persist the changes and inform the other swap participants about it in real time.
+
+### Example use case 2
+A user of the initialized wallet `swap-website-wallet` calls the `atomic-swap/tx-proposal` passing to it, besides all the necessary parameters to initialize a proposal, the new parameters `service.is_new` and `service.password`. This will make the headless app also initialize this proposal on the service and add it to its local `listenedProposals`.
+
+These identifier and password can then be shared with the other interested participants and future interactions can follow just like on example 1.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+### Listened proposals
+This integration with the swap service backend needs a way for the headless to store in memory the proposals that are currently being monitored by each of its initialized wallets.
+
+There are some considerations about this object:
+- An event listener will be created for each proposal to receive web-socket updates from the service
+- More than one initialized wallet can be listening to the same proposal
+  - There should be no more than one websocket connection open for each proposal
+- Wallets that did not register/create a proposal should not receive updates about it
+
+To achieve this, a `listenedProposals` map will be created on the `services/wallets.service` module containing, in short, the proposal identifier, password, and a list of wallet-ids that are listening to it:
+```ts
+{
+	proposalId: string,
+	password: string,
+	wallets: string[],
+}
+```
```

Comment:
```ts
interface txProposalConfig {
  id: string,
  password: string,
}

// keys are proposal ids
type walletProposals = Map<string, txProposalConfig>;

// keys are wallet-ids
const proposals = new Map<string, walletProposals>();
```

We don't need to create the intermediate type, I just used it for clarification.
The idea is that if we need to get all the proposals this wallet is listening this would be faster.
And we will need this operation since we may want to stop listening to all proposals of a wallet when the wallet is stopped.

Reply:
BTW, I think we need an api to list current proposals we are listening.
For instance, if a user wants to unregister a proposal but lost the id, he could list the proposals and unregister.

Reply:
Good suggestion, changed the document to add it. Also added a route for the list of `listenedProposals`